cmd/genkey.go

```diff
@@ -157,3 +186,65 @@ func generateAuthKeys(guid efiutil.EFIGUID, keyPath, keyType string) error {
 
 	return nil
 }
+
+// prepareCustomCertsDir takes a cert directory with keys as they are exported
+// from the UEFI firmware and prepares them for use with sbctl.
+// The keys are exported in the "authenticated variables" format.
+// The keys are expected to be in the "der" format in a specific directory structure.
+// The given directory should have the following files:
+// - db
+// - dbx
+// - KEK
+// - PK
+// It returns the prepared temporary directory where the keys are stored in
+// "der" format in the expected directories.
+func prepareCustomCertsDir(l v1.Logger) (string, error) {
+	customCertDir := viper.GetString(customCertDirFlag)
+	if customCertDir != "" {
+		if _, err := os.Stat(customCertDir); os.IsNotExist(err) {
+			return "", fmt.Errorf("custom cert directory does not exist: %s", customCertDir)
+		}
+	}
+
+	// create a temporary directory to store the custom certs
+	tmpDir, err := os.MkdirTemp("", "sbctl-custom-certs-*")
+	if err != nil {
+		return "", fmt.Errorf("creating temporary directory: %w", err)
+	}
+
+	// TODO: PK too?
+	for _, keyType := range []string{"db", "KEK"} {
+		b, err := ioutil.ReadFile(filepath.Join(customCertDir, keyType))
+		if err != nil {
+			return "", fmt.Errorf("reading custom cert file %s: %w", keyType, err)
+		}
+		f := bytes.NewReader(b)
+		siglist, err := signature.ReadSignatureDatabase(f)
+		if err != nil {
+			return "", fmt.Errorf("reading signature database: %w", err)
+		}
+
+		l.Infof("Converting custom certs (type: %s)\n", keyType)
+		for _, sig := range siglist {
```

Comment:
Code inspired by this: https://github.com/jimmykarily/go-uefi/blob/817a745e790a676138958ccc0e3c0e849d8c5f2a/cmd/efianalyze/main.go#L147